compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -401,23 +401,55 @@ LogicalResult getStaticDims(Operation *op,
   return success();
 }
 
+/// Utility to get the static sizes and strides for `AIEX::NpuDmaMemcpyNdOp`
+/// with implicit addressing.
+LogicalResult getStaticDimsForImplicitAddressing(
+    Operation *op, MemRefType memrefType, SmallVectorImpl<int64_t> &staticSizes,
+    SmallVectorImpl<int64_t> &staticStrides) {
+  // 1. Static sizes.
+  SmallVector<int64_t> shapeArr = llvm::to_vector(memrefType.getShape());
+  std::copy(shapeArr.begin(), shapeArr.end(),
+            staticSizes.end() - shapeArr.size());
```

Comment:
I think we should check that `staticSizes` is large enough. And it would be good to add a test case for this as well.